Flake unit test

We have a flaky unit test which might be just the way we have setup the unit test or an actual problem with how we generate the series, where labels are expected to be in one order but we get them in reverse. In any case we should make it be a deterministic order.

Example: https://github.com/kubernetes/kube-state-metrics/runs/1181030112

> --- FAIL: TestFullScrapeCycle (1.07s)
>    main_test.go:379: expected:   
 >       kube_pod_labels{namespace="default",pod="pod0"} 1, but got:
  >      kube_pod_labels{pod="pod0",namespace="default"} 1

--- a/pkg/allow/allow_labels.go
+++ b/pkg/allow/allow_labels.go
@@ -49,7 +49,7 @@ func (a Labels) Allowed(metric string, labels, values []string) ([]string, []str
 
 	var finalLabels, finalValues []string
 	labelSet := labelSet(allowedLabels)
-	for allowedLabel := range labelSet {
+	for _, allowedLabel := range labelSet {
 		for i, label := range labels {
 			if label == allowedLabel {
 				finalLabels = append(finalLabels, label)
@@ -61,12 +61,16 @@ func (a Labels) Allowed(metric string, labels, values []string) ([]string, []str
 	return finalLabels, finalValues
 }
 
-func labelSet(lists ...[]string) map[string]interface{} {
+func labelSet(lists ...[]string) []string {
 	m := make(map[string]interface{})
+	var set []string
 	for _, list := range lists {
 		for _, e := range list {
-			m[e] = struct{}{}
+			if _, ok := m[e]; !ok {
+				m[e] = struct{}{}
+				set = append(set, e)
+			}
 		}
 	}
-	return m
+	return set
 }

